### PR TITLE
[Semi-Modular] Stops Jacob's Ladder From Piercing Space Ruins

### PR DESCRIPTION
**Pull request by unit0016**
## About The Pull Request
Introduces a new global proc, "is_spaceruins_level()", and uses it to stop jacobs' ladder from.. this.
image
This has happened way too much to ignore it.

## Why It's Good For The Game
kinda hard to maintain a "do not fuck with each other"  policy when a magical ladder from snakes and ladders can completely cuck it

## Changelog
:cl:
balance: Jacob's ladder can no longer reach space levels that aren't empty. It can still reach lavaland and the station, however.
/:cl:

### CODE REVIEW

**Comment by jjpark-kb** [comment]
code is good, fix conflicting files.

File: code/__HELPERS/level_traits.dm [hunk 1]
```diff
@@ -10,3 +10,7 @@
 #define is_reserved_level(z) SSmapping.level_trait(z, ZTRAIT_RESERVED)
 
 #define is_away_level(z) SSmapping.level_trait(z, ZTRAIT_AWAY)
+
+//SKYRAT EDIT BEGIN//
+#define is_spaceruins_level(z) SSmapping.level_trait(z, ZTRAIT_SPACE_RUINS) //Somewhat of a cheesy way to go about this but it works.
```
Comment (Funce) [inline code/__HELPERS/level_traits.dm]: Please put this helper inside `/code/__HELPERS/~skyrat_helpers/level_traits.dm`

**Review by Funce — COMMENTED** [review]
If the ladder doesn't go to centcom, a reserved level, an away level, or a spaceruin level...... what else is there left for it to go on?

Do you know?

**Comment by unit0016** [comment]
> If the ladder doesn't go to centcom, a reserved level, an away level, or a spaceruin level...... what else is there left for it to go on?
> 
> Do you know?

Yeah. The station. Lavaland. Any space levels that don't have ruins. 

I was considering outright removing it completely.

**Comment by unit0016** [comment]
@Funce requested change done. Merge. Speedmerge, even. If I hear one more story about a miner having a ladder that can zoop into the DS-2 armory and uses it to tide the entire place for new-cops tier gear without issue because it's now directly connected to the station, I'm going to lose it.

**Pull request merged**